quartz/components/scripts/graph.inline.ts

```diff
@@ -100,7 +114,9 @@ async function renderGraph(container: string, fullSlug: FullSlug) {
     if (showTags) tags.forEach((tag) => neighbourhood.add(tag))
   }
 
-  const graphData: { nodes: NodeData[]; links: LinkData[] } = {
+  // XXX: How does links got morphed into LinkNodes here?
+  // links => LinkData[], where as links.filter(l => neighbourhood.has(l.source) && neighbourhood.has(l.target)) => LinkNodes[]
+  const graphData: { nodes: NodeData[]; links: LinkNodes[] } = {
```

Comment:
also do you know why links got morphed here?

iirc `array.filter` returns a reference of the original array, but we don't do any update or morph the link item to `links.source`?